### PR TITLE
Gun spinning 2.0

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request

This a refinement of gun spinning.
If you did not know, wearing a holster currently allows you to spin guns by using them in-hand while you reload

- The sprite actually spins now.

- Spinning is done by alt-clicking the gun

- Ballistic weapons are the only weapons that can be spun (_this includes rocket launchers_)

- Other actions that a gun may do by Alt-clicking happen simultaneously, except reskinning a gun such as the detective's revolver which takes priority over spinning.

- Clumsy people might accidentally shoot themselves while spinning guns (_this includes rocket launchers_)

## Why It's Good For The Game

My original draft of gun spinning was inspired but poorly thought out. This is much better.

## Changelog
:cl: itseasytosee
tweak: Major improvements to gun-spinning. It is now performed with an alt-click rather than using it in-hand.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
why not just make it a ctrl click option when held so it never conflicts?

**Comment by Pumpkinoe** [comment]
if ctrl click doesn't work either you could probably even make it a rightclick dropdown

**Comment by itseasytosee** [comment]
Currently, the only thing that conflicts is reskinning which I think should be prioritized over flipping. I could make it a verb.

**Comment by Jared-Fogle** [comment]
> you could probably even make it a rightclick dropdown

nonononono

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
You have generated new variables/procs/defines without proper documentation for the autodoc to work.

Please click [here ](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)for a guide on how to properly document these items.

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -331,9 +331,28 @@
...
+				user.visible_message(user, "<span class='notice'>Woah! [user] spins the [src] around their finger flinging off the [S].</span>")
+				gun_spin(user)
+			else
+				to_chat(user, "<span class='notice'>You unscrew \the [S] from \the [src].</span>")
 			user.put_in_hands(S)
 			clear_suppressor()
+			return
+	gun_spin(user)
+
+/obj/item/gun/ballistic/proc/gun_spin(mob/user)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/guns/ballistic.dm:344]: review

File: code/modules/projectiles/guns/ballistic.dm [hunk 2]
```diff
@@ -331,9 +331,28 @@
...
+	if(HAS_TRAIT(user, TRAIT_GUNFLIP) && (flip_cooldown <= world.time))
+		src.SpinAnimation(7,1)
+		if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(40))
+			to_chat(user, "<span class='userdanger'>While trying to flip the [src] you pull the trigger and accidently shoot yourself!</span>")
+			var/flip_mistake = pick(BODY_ZONE_L_LEG, BODY_ZONE_R_LEG, BODY_ZONE_HEAD, BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_CHEST)
+			process_fire(user, user, FALSE, flip_mistake)
+			user.dropItemToGround(src, TRUE)
+			return
+		flip_cooldown = (world.time + 30)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/guns/ballistic.dm:353]: time define

File: code/modules/projectiles/guns/ballistic.dm [hunk 3]
```diff
@@ -331,9 +331,28 @@
...
+		user.visible_message("<span class='notice'>[user] spins the [src] around their finger by the trigger. That’s pretty badass.</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/guns/ballistic.dm:354]:
```suggestion
		user.visible_message("<span class='notice'>[user] spins /the [src] around their finger by the trigger. That’s pretty badass.</span>")
```

account for proper nouns

File: code/modules/projectiles/guns/ballistic.dm [hunk 4]
```diff
@@ -331,9 +331,28 @@
...
+				gun_spin(user)
+			else
+				to_chat(user, "<span class='notice'>You unscrew \the [S] from \the [src].</span>")
 			user.put_in_hands(S)
 			clear_suppressor()
+			return
+	gun_spin(user)
+
+/obj/item/gun/ballistic/proc/gun_spin(mob/user)
+	if(HAS_TRAIT(user, TRAIT_GUNFLIP) && (flip_cooldown <= world.time))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/guns/ballistic.dm:345]:
```DM
if(!check)
    return
//rest of code here
```

File: code/modules/projectiles/guns/ballistic.dm [hunk 5]
```diff
@@ -351,7 +351,7 @@
 			user.dropItemToGround(src, TRUE)
 			return
 		flip_cooldown = (world.time + 30)
-		user.visible_message("<span class='notice'>[user] spins the [src] around their finger by the trigger. That’s pretty badass.</span>")
+		user.visible_message("<span class='notice'>[user] spins /the [src] around their finger by the trigger. That’s pretty badass.</span>")
```
Comment (Jared-Fogle) [inline code/modules/projectiles/guns/ballistic.dm]: so close, but also the is added automatically anyway

```suggestion
		user.visible_message("<span class='notice'>[user] spins [src] around their finger by the trigger. That’s pretty badass.</span>")
```

**Comment by Misdoubtful** [comment]
Imagine spinning your gun and the mag flies out and lands in your opponents hand.

File: code/modules/projectiles/guns/ballistic.dm (hunk 5 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/projectiles/guns/ballistic.dm]: i always forget if it defaults to a or the reeeeee

**Pull request closed without merging**